src/EditorFeatures/Core/Implementation/Classification/Semantic/AbstractSemanticOrEmbeddedClassificationViewTaggerProvider.cs

```diff
@@ -32,35 +29,33 @@ namespace Microsoft.CodeAnalysis.Editor.Implementation.Classification
     /// in the editor.  We use a view tagger so that we can only classify what's in view, and not
     /// the whole file.
     /// </summary>
-    [Export(typeof(IViewTaggerProvider))]
-    [TagType(typeof(IClassificationTag))]
-    [ContentType(ContentTypeNames.RoslynContentType)]
-    internal partial class SemanticClassificationViewTaggerProvider : AsynchronousViewTaggerProvider<IClassificationTag>
+    internal abstract class AbstractSemanticOrEmbeddedClassificationViewTaggerProvider : AsynchronousViewTaggerProvider<IClassificationTag>
     {
         private readonly ClassificationTypeMap _typeMap;
         private readonly IGlobalOptionService _globalOptions;
+        private readonly ClassificationType _type;
```

Comment:
this is the crux of the change.  there are now two taggers.  they operate independently, but almost identically.  The only difference is the type of classifications they request, which is specified by this type.